src/ShallowTraversal.js

```diff
@@ -29,7 +29,7 @@ export function childrenOfNode(node) {
 
 export function hasClassName(node, className) {
   const classes = propsOfNode(node).className || '';
-  return ` ${classes} `.indexOf(` ${className} `) > -1;
+  return ` ${classes.trim()} `.indexOf(` ${className} `) > -1;
```

Comment:
At least in Chrome the new line characters are perserved (if I check `node.className` on an element with the class `foo\nbar` it prints with the new line) but I'm not entirely sure how consistent that is.

But you're right, checking `classList` does still parse it as two classes as it should. For some reason I thought we were talking about parsing `foo\nbar` as a singular class name. I'll add update the PR to handle that case as well.